core/types/tx_rip7560.go

```diff
@@ -140,8 +140,38 @@ func callDataCost(data []byte) uint64 {
 	return nz*params.TxDataNonZeroGasEIP2028 + z*params.TxDataZeroGas
 }
 
-func (tx *Rip7560AccountAbstractionTx) CallDataGasCost() (uint64, error) {
-	return sumGas(callDataCost(tx.DeployerData), callDataCost(tx.ExecutionData), callDataCost(tx.PaymasterData))
+func (tx *Rip7560AccountAbstractionTx) PreTransactionGasCost() (uint64, error) {
+	calldataGasCost, err := tx.callDataGasCost()
+	if err != nil {
+		return 0, err
+	}
+	accessListGasCost := tx.accessListGasCost()
+	eip7702CodeInsertionsGasCost := tx.eip7702CodeInsertionsGasCost()
+	return params.Rip7560TxGas + calldataGasCost + accessListGasCost + eip7702CodeInsertionsGasCost, nil
+}
+
+func (tx *Rip7560AccountAbstractionTx) callDataGasCost() (uint64, error) {
+	return sumGas(
```

Comment:
nitpic: can safely ignore the error: all inputs are known to be well below 2^62, as they are based on actual TX size, not values in the tx tself.